modules/stereo/src/descriptor.hpp

```diff
@@ -0,0 +1,390 @@
+                                    c2 = c2 * 2;
+                                }
+                            }
+                        }
+                        dst1[(rWidth + j)] = c;
+                        dst2[(rWidth + j)] = c2;
+                    }
+                }
+            }
+        };
+        //!class that implemented the census descriptor on single images
+        class singleImageCensus : public ParallelLoopBody
+        {
+        private:
+            uint8_t *image;
+            int *dst;
+            int n2, width, height, type;
+        public:
+            singleImageCensus(uint8_t * img1, int w, int h, int k2, int * distance1,const int t) :
+                image(img1), dst(distance1), n2(k2), width(w), height(h), type(t){}
+            virtual void operator()(const cv::Range &r) const {
+                for (int i = r.start; i <= r.end ; i++)
+                {
+                    int rWidth = i * width;
+                    for (int j = n2; j <= width - n2; j++)
+                    {
+                        if (type == CensusImage::SSE)
+                        {
+                            //to do
+                        }
+                        else
+                        {
+                            int c = 0;
+                            for (int ii = i - n2; ii <= i + n2; ii++)
+                            {
+                                int rrWidth = ii * width;
+                                for (int jj = j - n2; jj <= j + n2; jj++)
+                                {
+                                    if (ii != i || jj != j)
+                                    {
+                                        if (image[(rrWidth + jj)] > image[(rWidth + j)])
+                                        {
+                                            c = c + 1;
+                                        }
+                                        c = c * 2;
+                                    }
+                                }
+                            }
+                            dst[(rWidth + j)] = c;
+                        }
+                    }
+                }
+            }
+        };
+        //! parallel implementation of MCT type of descriptors
+        class parallelMctDescriptor:public ParallelLoopBody
+        {
+        private:
+            uint8_t *image1, *image2;
+            int *dst1, *dst2;
+            int n2,t , width, height, type;
+        public:
+            parallelMctDescriptor(uint8_t * img1, uint8_t * img2, int w, int h, int k2,int threshold, int * distance1, int * distance2,const int tip) :
+                image1(img1), image2(img2), dst1(distance1), dst2(distance2), n2(k2), t(threshold), width(w), height(h), type(tip){}
+            virtual void operator()(const cv::Range &r) const {
+                for (int i = r.start; i <= r.end ; i++)
+                {
+                    int rWidth = i * width;
+                    int distV = (i)* width;
```

Comment:
No error, I forgot to delete it after refactoring the padding part
 


     On Tuesday, July 7, 2015 10:32 AM, Sergei Nosov <notifications@github.com> wrote:
   

 In modules/stereo/src/descriptor.hpp:> +        };
> +        //! parallel implementation of MCT type of descriptors
> +        class parallelMctDescriptor:public ParallelLoopBody
> +        {
> +        private:
> +            uint8_t *image1, *image2;
> +            int *dst1, *dst2;
> +            int n2,t , width, height, type;
> +        public:
> +            parallelMctDescriptor(uint8_t * img1, uint8_t * img2, int w, int h, int k2,int threshold, int * distance1, int * distance2,const int tip) :
> +                image1(img1), image2(img2), dst1(distance1), dst2(distance2), n2(k2), t(threshold), width(w), height(h), type(tip){}
> +            virtual void operator()(const cv::Range &r) const {
> +                for (int i = r.start; i <= r.end ; i++)
> +                {
> +                    int rWidth = i * width;
> +                    int distV = (i)* width;
It seems like rWidth == distV is always true. Is there an error?—
Reply to this email directly or view it on GitHub.